### ISSUE TITLE
Run-BCPTTestsInBcContainer fails at app compilation due to protection level

### ISSUE THREAD

**Issue by MJECloud** [microsoft/navcontainerhelper#2242]
**Describe the issue**
Hi Freddy, 

we want to run the PerformanceToolKit with Run-BCPTTestsInBcContainer. 
But the command fails after trying to compile the BCPTLogEntryAPI  extension. 

We're using a sandbox container with BC 19.1-DE

**Scripts used to create container and cause the issue**
```
Run-BCPTTestsInBcContainer -containerName BC -tenant default -companyName 'CRONUS DE' -credential (New-Object -TypeName pscredential -ArgumentList $Username, (ConvertTo-SecureString -String $Password -AsPlainText -Force)) -suiteCode TEST -Verbosey
```

**Full output of scripts**
```
Run-BCPTTestsInBcContainer -containerName BC -tenant default -companyName 'CRONUS DE' -credential (New-Object -TypeName pscredential -ArgumentList $Username, (ConvertTo-SecureString -String $Password -AsPlainText -Force)) -suiteCode TEST -Verbose
Adding BCPTLogEntryAPI.app to extend existing Performance Toolkit with BCPTLogEntry API page
Using Object Id 88123 (set $bcContainerHelperConfig.ObjectIdForInternalUse to change)
Using Symbols Folder: C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b\.alpackages
Disabling SSL Verification
Downloading symbols: Microsoft_Application_19.1.31886.34078.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=Application&versionText=18.0.0.0&tenant=default
Adding dependency to System Application from Microsoft
Adding dependency to Base Application from Microsoft
Downloading symbols: Microsoft_System_19.0.31889.34054.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=System&versionText=18.0.0.0&tenant=default
Downloading symbols: Microsoft_Performance Toolkit_18.0.0.0.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=Performance%20Toolkit&versionText=18.0.0.0&tenant=default
Adding dependency to Test Runner from Microsoft
Downloading symbols: Microsoft_System Application_19.1.31886.34078.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=System%20Application&versionText=19.1.0.0&tenant=default
Downloading symbols: Microsoft_Base Application_19.1.31886.34078.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=Base%20Application&versionText=19.1.0.0&tenant=default
Downloading symbols: Microsoft_Test Runner_19.1.31886.34078.app
Url : https://172.29.68.136:7049/BC/dev/packages?publisher=Microsoft&appName=Test%20Runner&versionText=19.1.0.0&tenant=default
Re-enabling SSL Verification
Compiling...
.\alc.exe /project:"C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b" /packagecachepath:"C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b\.alpackages" /out:"C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b\output\BcContainerHelper_BCPTLogEntryAPI_1.0.0.0.app" /assemblyprobingpaths:"C:\Program Files (x86)\Microsoft Dynamics NAV\190\RoleTailored Client","C:\Program Files\Microsoft Dynamics NAV\190\Service","C:\Program Files (x86)\Open XML SDK\V2.5\lib","c:\Windows\Microsoft.NET\Assembly","C:\Test Assemblies\Mock Assemblies"
Microsoft (R) AL Compiler version 8.2.8.28400
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'BCPTLogEntryAPI' containing '1' files at '10:11:34.290'.

C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b\BCPTLogEntryAPI.Page.al(15,19): error AL0161: 'BCPT Log Entry' is inaccessible due to its protection level
C:\ProgramData\BcContainerHelper\Extensions\BC\869aa3db-1d23-4864-8f3c-fb05adb14d1b\BCPTLogEntryAPI.Page.al(37,37): error AL0161: 'BCPT Header' is inaccessible due to its protection level

Compilation ended at '10:11:39.326'.

App generation failed with exit code 1
Compile-AppInBcContainer Telemetry Correlation Id: 947c9d6c-4809-4e37-af93-5c53e8207ec7
Run-BCPTTestsInBcContainer Telemetry Correlation Id: b0ae4661-8673-4c0b-a8a7-1728e734f112
App generation failed with exit code 1
```

**Comment by freddydk** [comment]
This code works fine on my machine:

```
$containerName = "bcserver"
$licenseFile = $LicenseFileSecret.SecretValue | Get-PlainText
$auth = "UserPassword"
$credential = New-Object pscredential -ArgumentList 'admin', $PasswordSecret.SecretValue
$artifactUrl = Get-BCArtifactUrl -country de -select Weekly

New-BcContainer `
    -containerName $containerName `
    -accept_eula `
    -artifact $artifactUrl `
    -auth $auth `
    -Credential $credential `
    -licenseFile $licenseFile `
    -updatehosts -includePerformanceToolkit

Run-BCPTTestsInBcContainer -containerName $containerName -tenant default -companyName 'CRONUS DE' -credential $credential -suiteCode TEST -Verbose
```

How did you create the container?
What script was used?
What version of ContainerHelper?

**Comment by freddydk** [comment]
I did a few fixes in BcContainerHelper 3.0.9 on performance testing, but this is an end-2-end script sample, running the samples app perf test and a sample suite:

```
$licenseFile = $LicenseFileSecret.SecretValue | Get-PlainText
$auth = "UserPassword"
$credential = New-Object pscredential -ArgumentList 'admin', $PasswordSecret.SecretValue
$containerName = "bcserver"
$countryCode = "de"
$artifactUrl = Get-BCArtifactUrl -country $countryCode -version 19

New-BCContainer -accept_eula -alwayspull `
                -containername $containername `
                -Auth UserPassword `
                -artifactUrl $artifactUrl `
                -Credential $credential `
                -shortcuts Desktop `
                -updateHosts `
                -includeTestToolkit -includeTestFrameworkOnly -includePerformanceToolkit


if (Get-BcContainerPath -containerName $containerName -path ":C:\Applications.$countryCode\Microsoft_Performance Toolkit Samples_*.app") {
    Publish-BcContainerApp -containerName $containerName -appFile ":C:\Applications.$countryCode\Microsoft_Performance Toolkit Samples_*.app" -skipVerification -install -sync
}
else {
    Publish-BcContainerApp -containerName $containerName -appFile ":C:\Applications\testframework\performancetoolkit\Microsoft_Performance Toolkit Samples.app" -skipVerification -install -sync
    
}

$bcptSuite = @'
{
  "code": "10UserTest",
  "description": "10 User Test",
  "durationInMinutes": 5,
  "defaultMinimumUserDelayInMilliSeconds": 120,
  "oneDayCorrespondsToInMinutes": 10,
  "tag": "10UserTest",
  "bcptSuiteLines": [
    {
      "codeunitID": 149107,
      "numberOfSessions": 1,
      "description": "BCPT Detail Trial Bal. Report",
      "minimumUserDelay": 100,
      "maximumUserDelay": 1000,
      "delayBetweenIterations": 10,
      "delayType": "Fixed",
      "runInForeground": false
    },
    {
      "codeunitID": 149103,
      "numberOfSessions": 1,
      "description": "BCPT Create PO with N Lines",
      "minimumUserDelay": 100,
      "maximumUserDelay": 1000,
      "delayBetweenIterations": 5,
      "delayType": "Fixed",
      "runInForeground": false,
      "parameters": "Lines=10"
    },
    {
      "codeunitID": 149105,
      "numberOfSessions": 1,
      "description": "BCPT Create SQ with N Lines",
      "minimumUserDelay": 100,
      "maximumUserDelay": 1000,
      "delayBetweenIterations": 5,
      "delayType": "Fixed",
      "runInForeground": false,
      "parameters": "Lines=10"
    },
    {
      "codeunitID": 149104,
      "numberOfSessions": 1,
      "description": "BCPT Create SO with N Lines",
      "minimumUserDelay": 100,
      "maximumUserDelay": 1000,
      "delayBetweenIterations": 5,
      "delayType": "Fixed",
      "runInForeground": false,
      "parameters": "Lines=10"
    }
  ]
}
'@ | ConvertFrom-Json 

$results = Run-BCPTTestsInBcContainer -containerName bcserver -credential $credential -BCPTSuite $bcptSuite -connectFromHost
$results | Out-Host
$results | ConvertTo-Json -Depth 99 | Set-Content -Path c:\temp\bcptTestResults.json -Encoding UTF8
```
